maildir_deduplicate/deduplicate.py

```diff
@@ -236,46 +237,50 @@ def canonical_header_value(cls, header, value):
-
-            too_dissimilar = self.messages_too_dissimilar(
-                hash_key, sorted_messages_size)
-            if too_dissimilar == 'size':
-                self.sizes_too_dissimilar += 1
-                continue
-            elif too_dissimilar == 'diff':
-                self.diff_too_big += 1
-                continue
-            elif too_dissimilar is False:
-                pass
-            else:
-                raise ValueError(
-                    "Unexpected value {!r} for too_dissimilar".format(
-                        too_dissimilar))
-
-            self.duplicates += len(messages) - 1
-            self.sets += 1
-
-            if self.strategy in [OLDER, NEWER]:
-                self.removed += self.process_duplicate_set(
-                    sorted_messages_ctime)
-            else:
-                self.removed += self.process_duplicate_set(
-                    sorted_messages_size)
+            try:
+                # Skip unique mails.
+                if len(message_files) == 1:
+                    continue
+                messages = [(mf, read_mailfile(mf)) for mf in message_files]
+    
+                subject = messages[0][1].get('Subject', '')
+                subject, count = re.subn('\s+', ' ', subject)
+                logger.info("Subject: {}".format(subject))
+    
+                if self.strategy == OLDER:
+                    sorted_messages_ctime = self.time_sort(messages, False)
+                elif self.strategy == NEWER:
+                    sorted_messages_ctime = self.time_sort(messages, True)
+    
+                sorted_messages_size = self.size_sort(messages)
+    
+                too_dissimilar = self.messages_too_dissimilar(
+                    hash_key, sorted_messages_size)
+                if too_dissimilar == 'size':
+                    self.sizes_too_dissimilar += 1
+                    continue
+                elif too_dissimilar == 'diff':
+                    self.diff_too_big += 1
+                    continue
+                elif too_dissimilar is False:
+                    pass
+                else:
+                    raise ValueError(
+                        "Unexpected value {!r} for too_dissimilar".format(
+                            too_dissimilar))
+    
+                self.duplicates += len(messages) - 1
+                self.sets += 1
+    
+                if self.strategy in [OLDER, NEWER]:
+                    self.removed += self.process_duplicate_set(
+                        sorted_messages_ctime)
+                else:
+                    self.removed += self.process_duplicate_set(
+                        sorted_messages_size)
+            except:
+                exc_type, exc_value, exc_traceback = exc_info()
+                logger.error("\n# # # # # # # EXCEPTION # # # # # # #\nType: %s\nValue: %s\nFiles parsed: %s\n# # # # # # # # # # # # # # # # # # #\n", exc_type, exc_value, message_files, exc_info=True)
```

Comment:
This line is causing Travis CI to fail with:

    ./maildir_deduplicate/deduplicate.py:283:80: E501 line too long (202 > 79 characters)

This is easily fixable, e.g. something like

    logger.error(
        "\n# # # # # # # EXCEPTION # # # # # # #\n"
        "Type: %s\nValue: %s\nFiles parsed: %s\n"
        "# # # # # # # # # # # # # # # # # # #\n", 
        exc_type, exc_value, message_files, exc_info=True)

(untested) should do it.